### ISSUE TITLE
convolution._crop_output gives wrong shape for odd input shapes

### ISSUE THREAD

**Issue by ira7bar** [brandondube/prysm#3]
The following code:
```
import numpy as np
from prysm.convolution import _crop_output
source = np.zeros((5, 5))
new = np.zeros((10, 10))
cropped = _crop_output(source, new)
print(cropped.shape)
```
prints the result (6, 6), instead of the expected (5,5).

**Comment by brandondube** [comment]
Thanks for the catch!  Fixed in d528ab7126630a66a24a4f066d2b8ba6b94fd52b.

I went with the opposite choice on floor and ceil -- fft for even-length arrays is biased in coordinates towards the right-hand side (that is, there is one less sample's worth of distance away from the origin over there), so the ceil-floor choice avoids a shearing away from the origin.

FWIW, I also fixed a bug with supports on analytical convolvables in bce4520b96c1f03920b9bbee92023e51f4d0d256 - so you may want to install the dev branch from git.

**Comment by brandondube** [comment]
I ended up reversing my decision in 641dff9584e4cbe174a4d3f1fd8c917944bc21a0 -- I had my understanding of ifftshift and fftshfit backwards
